independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/Reflections.java

```diff
@@ -148,4 +197,84 @@ public static Object invokeMethod(Class<?> clazz, String name, Class<?>[] paramT
         }
     }
 
+    static final class MethodKey {
+
+        final Class<?> clazz;
+        final String methodName;
+        final Class<?>[] parameterTypes;
+        final int hashCode;
+
+        public MethodKey(Class<?> clazz, String methodName, Class<?>[] parameterTypes) {
+            this.clazz = clazz;
+            this.methodName = methodName;
+            this.parameterTypes = parameterTypes;
+            final int prime = 31;
+            int result = 1;
+            result = prime * result + Arrays.hashCode(parameterTypes);
+            result = prime * result + ((clazz == null) ? 0 : clazz.hashCode());
+            result = prime * result + ((methodName == null) ? 0 : methodName.hashCode());
+            this.hashCode = result;
```

Comment:
Maybe move that part to a `buildHashCode` method? :)

Reply:
I don't think it's necessary to create a separate method for this ;-).

Reply:
As you prefer. Looks good to me, then!